src/browser/api_protocol/api_handlers/interappbus.ts

```diff
@@ -0,0 +1,204 @@
+import * as apiProtocolBase from './api_protocol_base';
+import { InterApplicationBus } from '../../api/interappbus';
+import ofEvents from '../../of_events';
+import route from '../../../common/route';
+import {
+    Acker,
+    APIMessage,
+    Identity,
+    Subscriber
+} from '../../../shapes';
+
+enum SubScriptionTypes {
+    MESSAGE = 1,
+    SUB_ADDED,
+    SUB_REMOVED
+}
+
+const successAck = {
+    success: true
+};
+
+const interAppBusExternalApiMap = {
+    'publish-message': publishMessage,
+    'send-message': sendMessage,
+    'subscribe': subscribe,
+    'unsubscribe': unsubscribe,
+    'subscriber-added': subscriberAdded,
+    'subscriber-removed': subscriberRemoved
+};
+
+export function init(): void {
+    apiProtocolBase.registerActionMap(interAppBusExternalApiMap);
+}
+
+function unsubscribe(identity: Identity, message: APIMessage, ack: Acker): void {
+    const { payload } = message;
+    const { topic } = payload;
+    const sourceUuid = '*';
+    const sourceWindowName = '*';
+    const subscriptionArgs = [
+        topic,
+        identity.uuid,
+        sourceUuid,
+        sourceWindowName,
+        SubScriptionTypes.MESSAGE
+    ];
+
+    apiProtocolBase.removeSubscription(identity, ...subscriptionArgs);
+
+    if (apiProtocolBase.subscriptionExists(identity, ...subscriptionArgs)) {
+        //if it still has the subscription, it didn't emit subscriber removed event.
+        InterApplicationBus.emitSubscriberRemoved(identity, payload);
+    }
+
+    ack(successAck);
+}
+
+function subscribe(identity: Identity, message: APIMessage, ack: Acker): void {
+    const { uuid, name } = identity;
+    const { payload } = message;
+    const { topic, messageKey: subscribedMessageKey } = payload;
+    const sourceUuid = '*';
+    const sourceWindowName = '*';
+
+    const subscriptionCallback = (payload: any): void => {
+        const { messageKey: sentMessageKey } = payload;
+        const command = { action: 'process-message', payload };
+
+        // old subscribing to new
+        if (!subscribedMessageKey && (sentMessageKey === 'messageString')) {
+            command.payload.message = JSON.parse(payload[sentMessageKey]);
+        }
+
+        apiProtocolBase.sendToIdentity(identity, command);
+    };
+
+    const subscriptionArgs = [
```

Comment:
Should we define a tuple type for IAB subscription args? They always have the same shape and this is the only type `any` in this file